BensonWollmanWeselyZombieProject/src/main/java/com/mycompany/bensonwollmanweselyzombieproject/Main.java

```diff
@@ -24,5 +26,76 @@ public static void main() {
         //Start Fight
         human1.attack_enemy(zombie1);
         zombie1.attack_enemy(human1);
+         */
+
+
+        /**
+         * Generate number of each unit and display to user
+         */
+        int numberOfHumans = (int)((Math.random() * 10) + 1);
+        int numberOfZombies = (int)((Math.random() * 10) + 1);
+
+        if (numberOfHumans > 1) {
+            System.out.println("We have " + numberOfHumans + " survivors trying to make it to safety.");
+        } else {
+            System.out.println("We have " + numberOfHumans + " survivor trying to make it to safety.");
+        }
+        if (numberOfZombies > 1) {
+            System.out.println("But there are " + numberOfZombies + " zombies waiting for them.");
+        } else {
+            System.out.println("But there is " + numberOfZombies + " zombie waiting for them.");
+        }
+        /**
+         * Create ArrayList for each unit type then generate unit objects for each type each unit type and add to it's
+         * respective ArrayList
+         */
+        ArrayList<Zombie> zombieArrayList = new ArrayList<>();
+        for (int x = 0; x < numberOfZombies; x++) {
+            zombieArrayList.add(new Zombie());
+        }
+
+        ArrayList<Survivor> survivorArrayList = new ArrayList<>();
+        for (int x = 0; x < numberOfHumans; x++) {
+            survivorArrayList.add(new Survivor());
+        }
+
+
+        /**
+         * Battle
+         */
+        while (!zombieArrayList.isEmpty() && !survivorArrayList.isEmpty()) {
+            /**
+             * Survivors attack, remove defeated zombies from zombieArrayList
+             */
+            for (int x = 0; x < survivorArrayList.size(); x++) {
+                for (int y = 0; y < zombieArrayList.size(); y++) {
+                    survivorArrayList.get(x).attack_enemy(zombieArrayList.get(y));
+                    if (zombieArrayList.get(y).health < 1) {

```

Comment:
Lets use is_alive here, that's probably a follow along PR either Philly Dawggg or I can do since you supplied the bulk of it here.

Reply:
Completely agree, this should be using is_alive().